test/src/SpendPermissions/approve.t.sol

```diff
@@ -258,4 +260,104 @@ contract ApproveTest is SpendPermissionManagerBase {
         });
         mockSpendPermissionManager.approve(spendPermission);
     }
+
+    function test_approve_success_returnsTrue(
+        address account,
+        address spender,
+        uint48 start,
+        uint48 end,
+        uint48 period,
+        uint160 allowance,
+        uint256 salt,
+        bytes memory extraData
+    ) public {
+        vm.assume(spender != address(0));
+        vm.assume(start < end);
+        vm.assume(period > 0);
+        vm.assume(allowance > 0);
+
+        SpendPermissionManager.SpendPermission memory spendPermission = SpendPermissionManager.SpendPermission({
+            account: account,
+            spender: spender,
+            token: NATIVE_TOKEN,
+            start: start,
+            end: end,
+            period: period,
+            allowance: allowance,
+            salt: salt,
+            extraData: extraData
+        });
+        vm.prank(account);
+        bool approved = mockSpendPermissionManager.approve(spendPermission);
+        vm.assertTrue(approved);
+        vm.assertTrue(mockSpendPermissionManager.isApproved(spendPermission));
+    }
+
+    function test_approve_success_returnsTrueNoEventEmittedIfAlreadyApproved(
+        address account,
+        address spender,
+        uint48 start,
+        uint48 end,
+        uint48 period,
+        uint160 allowance,
+        uint256 salt,
+        bytes memory extraData
+    ) public {
+        vm.assume(spender != address(0));
+        vm.assume(start < end);
+        vm.assume(period > 0);
+        vm.assume(allowance > 0);
+
+        SpendPermissionManager.SpendPermission memory spendPermission = SpendPermissionManager.SpendPermission({
+            account: account,
+            spender: spender,
+            token: NATIVE_TOKEN,
+            start: start,
+            end: end,
+            period: period,
+            allowance: allowance,
+            salt: salt,
+            extraData: extraData
+        });
+        vm.startPrank(account);
+        mockSpendPermissionManager.approve(spendPermission); // approve permission before second approval
+        vm.recordLogs();
+        bool approved = mockSpendPermissionManager.approve(spendPermission);
+        Vm.Log[] memory logs = vm.getRecordedLogs();
+        vm.assertEq(logs.length, 0); // no event emitted
+        vm.assertTrue(approved);
+        vm.assertTrue(mockSpendPermissionManager.isApproved(spendPermission));
+    }
+
+    function test_approve_success_returnsFalseIfPermissionRevoked(
```

Comment:
To be extra thorough, we also want to check that if false is returned, then the permission is not approved and there was no event emitted.